### PR TITLE
[CCI] Update DEVELOPER_GUIDE.md.

### PR DESCRIPTION
**Pull request by roma2023**
### Description
add docker installation and run commands to DEVELOPER_GUIDE.md.

### Issues Resolved
issue #341 resolved

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -17,6 +17,16 @@ $ nox -rs format
 
 # Run the test suite
 $ nox -rs test
+
+```
+
+## Install and Run Docker Image
```
Comment (dblock) [inline DEVELOPER_GUIDE.md:23]: Update TOC please.

File: DEVELOPER_GUIDE.md [hunk 2]
```diff
@@ -17,6 +17,16 @@ $ nox -rs format
...
+Note that integration tests require docker to be installed and running, and downloads quite a bit of data from over the internet and hence take few minutes to complete.
```
Comment (dblock) [inline DEVELOPER_GUIDE.md:24]: Nit: I know you copy-pasted this from below, but let's keep it functional. I would just say "Integration tests require [docker](link ...)." The whole "it takes a few minutes ... " seems unnecessary.

Does one have to run the image before running tests? If not, say "Integration tests will auto-start the docker image. To start it manually:" or something like that.

**Review by dblock — CHANGES_REQUESTED** [review]
Looks like you have unrelated changes in this PR. The rest is me being nitpicky.

File: DEVELOPER_GUIDE.md (hunk 1 above)
Comment (roma2023) [reply DEVELOPER_GUIDE.md]: Done!!!

File: DEVELOPER_GUIDE.md (hunk 2 above)
Comment (roma2023) [reply DEVELOPER_GUIDE.md]: Thank you for your feedback. Done!

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
Can you please rebase/resolve conflicts. Make sure PR numbers match. Thanks!

**Review by VachaShah — APPROVED** [review]

**Comment by VachaShah** [comment]
@roma2023 Please resolve the conflicts on this PR by merging the latest changes from the main branch.

**Comment by dblock** [comment]
@roma2023 Looks like you're dragging in a bunch of other changes. 

https://code.dblock.org/2015/08/31/getting-out-of-your-first-git-mess.html might be helpful?

**Comment by saimedhi** [comment]
Hello, @roma2023. Thank you for your contribution to the opensearch-py repository. Please make the following modifications:

1. Add a change log entry.
2. DCO has not been signed. When committing, use the -s flag. For example `git commit -s -m "Updated Developer_Guide"`. 
3. Merge all commits into single commit. If you have trouble, close this PR and create a new PR.

Thanks.

**Comment by roma2023** [comment]
This PR was closed due to unknown "git mess," I have tried to tackle it in different ways but the issue was not gone. I will try to create a new PR with the up-to-date version of the master, I hope this will solve it. Please let me know if you have any questions regarding this PR. Thank you for your support!

**Comment by dblock** [comment]
Speaking of git mess, https://code.dblock.org/2015/08/31/getting-out-of-your-first-git-mess.html may be helpful.

**Pull request closed without merging**